packages/peregrine/lib/talons/CheckoutPage/AddressBook/useAddressBook.js

```diff
@@ -0,0 +1,138 @@
+        mutations: { setCustomerAddressOnCartMutation },
+        queries: { getCustomerAddressesQuery, getCustomerCartAddressQuery },
+        toggleActiveContent
+    } = props;
+
+    const [, { toggleDrawer }] = useAppContext();
+    const [{ cartId }] = useCartContext();
+
+    const addressCount = useRef();
+    const [activeAddress, setActiveAddress] = useState();
+    const [selectedAddress, setSelectedAddress] = useState();
+
+    const [
+        setCustomerAddressOnCart,
+        { loading: setCustomerAddressOnCartLoading }
+    ] = useMutation(setCustomerAddressOnCartMutation);
+
+    const {
+        data: customerAddressesData,
+        error: customerAddressesError,
+        loading: customerAddressesLoading
+    } = useQuery(getCustomerAddressesQuery);
+
+    const {
+        data: customerCartAddressData,
+        error: customerCartAddressError,
+        loading: customerCartAddressLoading
+    } = useQuery(getCustomerCartAddressQuery);
+
+    useEffect(() => {
+        if (customerAddressesError) {
+            console.error(customerAddressesError);
+        }
+
+        if (customerCartAddressError) {
+            console.error(customerCartAddressError);
+        }
+    }, [customerAddressesError, customerCartAddressError]);
+
+    const isLoading =
+        customerAddressesLoading ||
+        customerCartAddressLoading ||
+        setCustomerAddressOnCartLoading;
+
+    const customerAddresses =
+        (customerAddressesData && customerAddressesData.customer.addresses) ||
+        [];
+
+    useEffect(() => {
+        if (customerAddresses.length !== addressCount.current) {
+            // Auto-select newly added address when count changes
+            if (addressCount.current) {
+                const newestAddress =
+                    customerAddresses[customerAddresses.length - 1];
+                setSelectedAddress(newestAddress.id);
+            }
+
+            addressCount.current = customerAddresses.length;
+        }
+    }, [customerAddresses]);
+
+    const handleEditAddress = useCallback(
+        address => {
+            setActiveAddress(address);
+            toggleDrawer('shippingInformation.edit');
+        },
+        [toggleDrawer]
+    );
+
+    const handleAddAddress = useCallback(() => {
+        handleEditAddress();
+    }, [handleEditAddress]);
+
+    const handleSelectAddress = useCallback(addressId => {
+        setSelectedAddress(addressId);
+    }, []);
+
+    if (
+        customerAddresses.length &&
+        customerCartAddressData &&
+        !selectedAddress
+    ) {
+        const { customerCart } = customerCartAddressData;
+        const { shipping_addresses: shippingAddresses } = customerCart;
+        if (shippingAddresses.length) {
+            const primaryCartAddress = shippingAddresses[0];
+
+            const foundSelectedAddress = customerAddresses.find(
+                customerAddress =>
+                    customerAddress.street[0] === primaryCartAddress.street[0]
```

Comment:
Is there another way we could find a match? For example I have two addresses saved in my address book - one for me and one for my wife but our street is the same. How would that work here?

Reply:
Wouldn't it be great if GraphQL just returned us this information 🙄 I'm willing to add first/last to the heuristic if you think that'll help.

Reply:
Yea I think first and last is sufficient. In the future you could go so far as to trim and hash the fields.

Reply:
Added in b5aec84.